scripts/utils/consensus-celltype-plotting-functions.R

```diff
@@ -0,0 +1,185 @@
+    ) 
+  
+  # no longer need this and it takes up space 
+  rm(consensus_dt)
+  gc()
+  
+  # get list of celltypes to keep and assign colors 
+  celltype_groups <- group_stats_df |> 
+    dplyr::pull(broad_celltype_group) |> 
+    unique() |>
+    as.character()
+  
+  # filter markers to those that are actually relevant 
+  # we will only plot the marker genes for cell types that are part of the assigned broad validation group for this group of samples
+  # we don't care about plotting marker genes for cell types that aren't present here 
+  filtered_markers_df <- markers_dt |> 
+    dplyr::filter(validation_group_annotation %in% celltype_groups,
+                  gene_symbol %in% group_stats_df$gene_symbol)
+
+  # specify x axis order for dotplot
+  marker_gene_order <- filtered_markers_df |> 
+    dplyr::pull(gene_symbol)
+  
+  # set order for cell types 
+  celltype_order <- unique(filtered_markers_df$validation_group_annotation)
+  
+  # filter out low expressed genes
+  dotplot_df <- group_stats_df |> 
+    dplyr::filter(mean_exp > 0, percent_exp > 10) |> 
+    dplyr::arrange(broad_celltype_group) |> 
+    # add a label for the plot 
+    dplyr::mutate(y_label = as.factor(glue::glue("{broad_celltype_group} ({total_cells})"))) |> 
+    # remove marker genes that aren't present in final annotations and set x axis order 
+    dplyr::filter(gene_symbol %in% marker_gene_order) |> 
+    dplyr::mutate(
+      # set orders of gene symbol and validation groups 
+      gene_symbol = factor(gene_symbol, levels = marker_gene_order),
+      validation_group_annotation = factor(validation_group_annotation, levels = celltype_order)
+    )
+  
+  
+  # make dotplot with marker gene exp
+  dotplot <- ggplot(dotplot_df, aes(y = forcats::fct_rev(y_label), x = gene_symbol, color = mean_exp, size = percent_exp)) +
+    geom_point() +
+    scale_color_viridis_c(option = "magma") +
+    facet_grid(cols = vars(validation_group_annotation), scales = "free", space = "free") +
+    theme_classic() +
+    theme(
+      strip.background = element_blank(),
+      strip.text.x = element_blank(), 
+      axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5),
+      axis.ticks.x = element_blank(),
+      text = element_text(size = 14),
+      panel.spacing = unit(0.5, "lines") # adjust spacing and match with annotation bar
+    ) +
+    labs(
+      x = "",
+      y = "Broad cell type annotation",
+      color = "Mean gene expression",
+      size = "Percent cells expressed"
+    ) 
+  
+  
+  # add annotation bar aligning marker genes with validation group 
+  color_bar <- ggplot(dotplot_df, aes(x = gene_symbol, y = 1, fill = validation_group_annotation)) + 
+    geom_tile() + 
+    facet_grid(cols = vars(validation_group_annotation), scales = "free", space = "free", switch = "x") +
+    scale_fill_manual(values = celltype_colors, breaks = levels(dotplot_df$validation_group_annotation)) +
+    ggmap::theme_nothing() +
```

Comment:
Curiosity (and dependency) question: how is this different from `ggplot2::theme_void()`?